tests/storage/test_state.py

```diff
@@ -606,23 +610,29 @@ def test_batched_state_group_storing(self) -> None:
         # has an entry and prev event in state_group_edges
         for event, context in processed_events_and_context:
             if event.is_state():
-                state = self.get_success(
-                    self.store.db_pool.simple_select_list(
-                        table="state_groups_state",
-                        keyvalues={"state_group": context.state_group_after_event},
-                        retcols=("type", "state_key"),
-                    )
-                )
-                self.assertEqual(event.type, state[0].get("type"))
-                self.assertEqual(event.state_key, state[0].get("state_key"))
-
-                groups = self.get_success(
-                    self.store.db_pool.simple_select_list(
-                        table="state_group_edges",
-                        keyvalues={"state_group": str(context.state_group_after_event)},
-                        retcols=("*",),
-                    )
+                state = cast(
+                    List[Tuple[str, str]],
```

Comment:
Also nullable in the DB. Maybe we don't care for tests though :shrug:

Reply:
Sorry -- what's nullable here? I'm not seeing that.

Reply:
Everything in state_groups_state, AFAICS:

```
matrix=> \d state_groups_state
           Table "matrix.state_groups_state"
   Column    │  Type  │ Collation │ Nullable │ Default 
═════════════╪════════╪═══════════╪══════════╪═════════
 state_group │ bigint │           │          │ 
 room_id     │ text   │           │          │ 
 type        │ text   │           │          │ 
 state_key   │ text   │           │          │ 
 event_id    │ text   │           │          │ 
Indexes:
    "state_groups_state_room_id_idx" brin (room_id) WITH (pages_per_range='1')
    "state_groups_state_type_idx" btree (state_group, type, state_key), tablespace "postgres_second"
```

Reply:
My local SQLite database doesn't allow null:

```sql
sqlite> PRAGMA table_info(state_groups_state);
┌─────┬─────────────┬────────┬─────────┬────────────┬────┐
│ cid │    name     │  type  │ notnull │ dflt_value │ pk │
├─────┼─────────────┼────────┼─────────┼────────────┼────┤
│ 0   │ state_group │ BIGINT │ 1       │            │ 0  │
│ 1   │ room_id     │ TEXT   │ 1       │            │ 0  │
│ 2   │ type        │ TEXT   │ 1       │            │ 0  │
│ 3   │ state_key   │ TEXT   │ 1       │            │ 0  │
│ 4   │ event_id    │ TEXT   │ 1       │            │ 0  │
└─────┴─────────────┴────────┴─────────┴────────────┴────┘
```

The [schema also doesn't allow it](https://github.com/matrix-org/synapse/blob/0a38c7ec6d46b6e51bfa53ff44e51637d3c63f5c/synapse/storage/schema/state/full_schemas/72/full.sql.postgres#L16-L22), seems like a matrix.org issue?

Reply:
(You can probably deduce how I lookup the schema)

Reply:
See #11893 also.